### PR TITLE
Refine excluded AMP prefix

### PR DESCRIPTION
**Pull request by danieldk**
Refine AMP function exclusion: `_amp` -> `_amp_foreach`

Exclusion of the `_amp` prefix also excludes `internal_amp_update_scale`,
which is needed for mixed-precision training.

### CODE REVIEW

**Review by LaurentMazare — APPROVED** [review]

**Comment by LaurentMazare** [comment]
Looks good, thanks for the PR! (though it would seem better if there is a function to do this that is not flagged as "internal" but it doesn't seem to be the case)

**Comment by lostmsu** [comment]
Is there a example for using AMP?

**Comment by joverwey** [comment]
I have the same question. How do you train with mixed precision. PyTorch has a class called `torch.cuda.amp.GradScalar`. What is the equivalent in `tch-rs`?

**Comment by danieldk** [comment]
Here's an example of gradient scaling in Rust:

https://github.com/tensordot/syntaxdot/blob/main/syntaxdot/src/optimizers/grad_scale.rs

You also have to use `autocast` in forward passes:

https://github.com/tensordot/syntaxdot/blob/25a0f1893ef8107b2510b05cf977533db06affee/syntaxdot-cli/src/util.rs#L32
